docs/design/mesos-style.md

```diff
@@ -0,0 +1,174 @@
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Building Mesos/Omega-style frameworks on Kubernetes
+
+## Introduction
+
+We have observed two different cluster management architectures, which can be categorized as "Borg-style" and "Mesos/Omega-style."
+(In the remainder of this document, we will abbreviate the latter as "Mesos-style.")
+Although out-of-the box Kubernetes uses a Borg-style architecture, it can also be configured in a Mesos-style architecture,
+and in fact can support both styles at the same time. This document describes the two approaches and describes how
+to deploy a Mesos-style architecture on Kubernetes.
+
+(As an aside, the converse is also true: one can deploy a Borg/Kubernetes-style architecture on Mesos.)
+
+This document is NOT intended to provide a comprehensive comparison of Borg and Mesos. For example, we omit discussion
+of the tradeoffs between scheduling with full knowledge of cluster state vs. scheduling using the "offer" model.
+(That issue is discussed in some detail in the Omega paper (see references section at the end of this doc).)
+
+
+## What is a Borg-style architecture?
+
+A Borg-style architecture is characterized by:
+* a single logical API endpoint for clients, where some amount of processing is done on requests, such as admission control and applying defaults
+* generic (non-application-specific) collection abstractions described declaratively,
+* generic controllers/state machines that manage the lifecycle of the collection abstractions and the containers spawned from them
+* a generic scheduler
+
+For example, Borg's primary collection abstraction is a Job, and every application that runs on Borg--whether it's a user-facing
+service like the GMail front-end, a batch job like a MapReduce, or an infrastructure service like GFS--must represent itself as
+a Job. Borg has corresponding state machine logic for managing Jobs and their instances, and a scheduler that's responsible
+for assigning the instances to machines.
+
+The flow of a request in Borg is:
+
+1. Client submits a collection object to the Borgmaster API endpoint
+1. Admission control, quota, applying defaults, etc. run on the collection
+1. If the collection is admitted, it is persisted, and the collection state machine creates the underlying instances
+1. The scheduler assigns a hostname to the instance, and tells the Borglet to start the instance's container(s)
+1. Borglet starts the container(s)
+1. The instance state machine manages the instances and the collection state machine manages the collection during their lifetimes
+
+Out-of-the-box Kubernetes has *workload-specific* abstractions (ReplicaSet, Job, DaemonSet, etc.) and corresponding controllers,
+and in the future may have [workload-specific schedulers](../../docs/proposals/multiple-schedulers.md),
```

Comment:
Are you sure? I wasn't aware of anything immediately planned, though I expect it will happen eventually.